internal/repos/gitlab.go

```
@@ -88,14 +98,29 @@ func newGitLabSource(svc *types.ExternalService, c *schema.GitLabConnection, cf
 
 	provider := gitlab.NewClientProvider(baseURL, cli)
 
+	client := provider.GetPATClient(c.Token, "")
+
+	dotcomMode := envvar.SourcegraphDotComMode()
+	if !dotcomMode || (dotcomMode && c.CloudGlobal) {
+		rlm := client.RateLimitMonitor()
+		if rlm != nil {
```

When would it be nil? Is that a valid state?

---

I guess it's not possible the way the code is at the moment. I'l remove the check since it'll fail hard if we change the code later so we'll spot it easily.